clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinator.java

```diff
@@ -623,24 +623,15 @@ public boolean commitOffsetsSync(Map<TopicPartition, OffsetAndMetadata> offsets,
         return false;
     }
 
-    private void maybeAutoCommitOffsetsAsync(long now) {
+    public void maybeAutoCommitOffsetsAsync(long now, Map<TopicPartition, OffsetAndMetadata> allConsumedOffsets) {
         if (autoCommitEnabled) {
-            if (coordinatorUnknown()) {
-                this.nextAutoCommitDeadline = now + retryBackoffMs;
-            } else if (now >= nextAutoCommitDeadline) {
-                this.nextAutoCommitDeadline = now + autoCommitIntervalMs;
-                doAutoCommitOffsetsAsync();
+            if (now >= nextAutoCommitDeadline) {
+                doAutoCommitOffsetsAsync(allConsumedOffsets);
             }
         }
     }
 
-    public void maybeAutoCommitOffsetsNow() {
-        if (autoCommitEnabled && !coordinatorUnknown())
-            doAutoCommitOffsetsAsync();
-    }
-
-    private void doAutoCommitOffsetsAsync() {
-        Map<TopicPartition, OffsetAndMetadata> allConsumedOffsets = subscriptions.allConsumed();
+    private void doAutoCommitOffsetsAsync(Map<TopicPartition, OffsetAndMetadata> allConsumedOffsets) {
```

Comment:
Why add this argument? As far as I can tell, all uses just pass `subscriptions.allConsumed()`, so the original was fine?